sdk/core/System.Net.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs

```diff
@@ -0,0 +1,225 @@
+                responseMessage = _httpClient.Send(httpRequest, HttpCompletionOption.ResponseHeadersRead, message.CancellationToken);
+#pragma warning restore CA1416
+            }
+            else
+#endif
+            {
+#pragma warning disable AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+                responseMessage = await _httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead, message.CancellationToken).ConfigureAwait(false);
+#pragma warning restore AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+            }
+
+            if (responseMessage.Content != null)
+            {
+#if NET5_0_OR_GREATER
+                if (async)
+                {
+                    contentStream = await responseMessage.Content.ReadAsStreamAsync(message.CancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    contentStream = responseMessage.Content.ReadAsStream(message.CancellationToken);
+                }
+#else
+#pragma warning disable AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+                contentStream = await responseMessage.Content.ReadAsStreamAsync().ConfigureAwait(false);
+#pragma warning restore AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+#endif
+            }
+        }
+        // HttpClient on NET5 throws OperationCanceledException from sync call sites, normalize to TaskCanceledException
+        catch (OperationCanceledException e) when (ClientUtilities.ShouldWrapInOperationCanceledException(e, message.CancellationToken))
+        {
+            throw ClientUtilities.CreateOperationCanceledException(e, message.CancellationToken);
+        }
+        catch (HttpRequestException e)
+        {
+            throw new UnsuccessfulRequestException(e.Message, e);
+        }
+
+        // This extensibility point lets derived types do the following:
+        //   1. Set message.Response to an implementation-specific type, e.g. Azure.Core.Response.
+        //   2. Make any necessary modifications based on the System.Net.Http.HttpResponseMessage.
+        OnReceivedResponse(message, responseMessage);
+
+        // We set derived values on the MessageResponse here, including Content and IsError
+        // to ensure these things happen in the transport.  If derived implementations need
+        // to override these default transport values, they can do so in pipeline policies.
+
+        // TODO: a possible alternative is to make instantiating the Response a specific
+        // extensibilty point and let OnReceivedResponse enable transport-specific logic.
+        // Consider which is preferred as part of holistic extensibility-point review.
+        if (contentStream is not null)
+        {
+            message.Response.ContentStream = contentStream;
+        }
+
+        message.Response.IsError = message.MessageClassifier.IsError(message);
```

Comment:
We have a bug here today where if a client-user or client-author overrides the transport, message.Response.IsError will never be set.

This isn't a problem in Azure.Core today because we have an internal TransportPolicy that sets response.IsError.  The issue is introduced by making the transport a policy itself.

Options here include:
- Adding a public TransportPolicy that could be used in both ClientModel clients and Azure.Core (we would make transport not be a policy anymore, I think?)
- Adding an internal policy to set response.IsError.  This could be:
  - an internal TransportPolicy that invokes the transport
  - a policy that sits right before the transport so IsError is set for all other policies - this would allow us to keep transport a policy if we wanted that.

The disadvantage of the latter approach is that we would be setting IsError differently in ClientModel and Azure.Core-based clients.